ARTris/Localization/LocalizationViewController.swift

```diff
@@ -0,0 +1,129 @@
+import UIKit
+import Foundation
+import ARKit
+import SceneKit
+
+class LocalizationViewController: UIViewController,ARSessionDelegate
+{
+    private var shadow: GridShadow!
+    private var shadowBinder: SCNNode!
+    private var objectPosition: CGPoint?
+    private var rotationRecognizer: UIRotationGestureRecognizer!
+    var sessionId: String!
+    
+    @IBOutlet weak var userPromptLabel: UILabel!
+    @IBOutlet weak var backBtn: UIButton!
+    @IBOutlet weak var sceneView: VirtualObjectARView! {
+        didSet {
+            let rotationHandler = #selector(LocalizationViewController.rotate(byReactingTo:))
+            rotationRecognizer = UIRotationGestureRecognizer(target: self, action: rotationHandler)
+            sceneView.addGestureRecognizer(rotationRecognizer)
+        }
+    }
+    @IBOutlet weak var nextBtn: UIButton! {
+        didSet {
+            nextBtn.isHidden = true
+        }
+    }
+    private lazy var gameViewController: GameViewController = {
+        let viewController = storyboard?.instantiateViewController(withIdentifier: "gameViewController") as! GameViewController
+        viewController.sceneView = sceneView
+        viewController.parentNode = shadowBinder
+        viewController.sessionId = sessionId
+        return viewController
+    }()
+    private var screenCentre: CGPoint {
+        let bounds = sceneView.bounds
+        return CGPoint(x: bounds.midX, y: bounds.midY)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        sceneView.session.delegate = self
+        sceneView.scene = SCNScene()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        let configuration = ARWorldTrackingConfiguration()
+        sceneView.session.run(configuration)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        shadowBinder = SCNNode(geometry: nil)
+        shadow = GridShadow(w: 8, l: 8, parent: shadowBinder, size: 0.02, color: UIColor.green)
```

Comment:
move the object literals to a struct or add them as default values to the initializer.